### ISSUE TITLE
Create `Task` for publishing Tekton Pipeline release yaml

### ISSUE THREAD

**Issue by bobcatfish** [tektoncd/pipeline#529]
## Expected behaviour

We should be testing and releasing Pipelines with Pipelines.

We should have a `Task` (and related Resources) defined that we can use to:
- Generate the `release.yaml` (including the correct image names)
- Upload the `release.yaml` to a gcs bucket

### The Task

* Inputs:
  - git repo
  - gcr images (for all Tekton images - we can assume these have been built by something upstream, specifically this is going to be #528)
* Outputs:
  - gcs bucket (where the release.yaml will be written)
* Parameters:
- ?

## Actual Behavior

We currently use [prow](https://github.com/knative/test-infra/tree/master/ci/prow) and [knative release scripts](https://github.com/knative/build-pipeline/blob/master/hack/release.md) for releasing.

## Additional Info

* Blocked by https://github.com/knative/build-pipeline/issues/216 (we have no way to get the digests of previously built images) - we could implement this assuming we have support for digests if we want (the field does exist in the PipelineResource)
* See [0.1 release.yaml](https://github.com/knative/build-pipeline/releases/download/v0.1.0/release.yaml) to see all images required
* Actually triggering this is not in scope for this issue, it's fine to test manually (we'll be linking these together in a later issue)
* PR should include docs that explain how the Task works
